Clean up C++ to eliminate warnings

To do:

- [x] chol_derivative.h:81:7: warning: ‘chols<double>::n_visits’ will be initialized after [-Wreorder]
   81 |   int n_visits;
      |       ^~~~~~~~
- [x] chol_derivative.h:79:15: warning:   ‘std::string chols<double>::cov_type’ [-Wreorder]
   79 |   std::string cov_type;
      |               ^~~~~~~~
- [x] chol_derivative.h:89:3: warning:   when initialized here [-Wreorder]
   89 |   chols(vector<Type> theta, int n_visits, std::string cov_type): n_visits(n_visits), cov_type(cov_type), full_visit(std::vector<int>(n_visits)) {
      |   ^~~~~
- [x] chol_derivative.h:152:11: warning: unused variable ‘n_visists_i’ [-Wunused-variable]
  152 |       int n_visists_i = visits.size();
      |           ^~~~~~~~~~~
- [x] utils.h:23:21: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<int>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
   23 |   for (int i = 0; i < visits_i.size(); i++) {
      |                   ~~^~~~~~~~~~~~~~~~~

--- a/src/chol_derivative.h
+++ b/src/chol_derivative.h
@@ -77,8 +77,8 @@ struct chols {
   std::map<std::vector<int>, matrix<Type>> sigma_inverse_d1_cache;
   std::map<std::vector<int>, Eigen::SparseMatrix<Type>> sel_mat_cache;
   std::string cov_type;
-  std::vector<int> full_visit;
   int n_visits;
+  std::vector<int> full_visit;
   int n_theta; // theta.size()
   vector<Type> theta;
   matrix<Type> chol_full;
@@ -86,7 +86,7 @@ struct chols {
     // this default constructor is needed because the use of `[]`.
   }
   // constructor from theta, n_visits and cov_type, and cache full_visits values
-  chols(vector<Type> theta, int n_visits, std::string cov_type): n_visits(n_visits), cov_type(cov_type), full_visit(std::vector<int>(n_visits)) {
+  chols(vector<Type> theta, int n_visits, std::string cov_type): cov_type(cov_type), n_visits(n_visits), full_visit(std::vector<int>(n_visits)) {
     this->theta = theta;
     for (int i = 0; i < n_visits; i++) {
       this->full_visit[i] = i;
@@ -149,7 +149,6 @@ struct chols {
       return this->sigma_cache[visits];
     } else {
       Eigen::SparseMatrix<Type> sel_mat = this->get_sel_mat(visits);
-      int n_visists_i = visits.size();
       matrix<Type> ret = sel_mat * this->sigma_cache[this->full_visit] * sel_mat.transpose();
       this->sigma_cache[visits] = ret;
       return ret;

--- a/src/utils.h
+++ b/src/utils.h
@@ -20,7 +20,7 @@ Eigen::SparseMatrix<Type> get_select_matrix(const vector<int>& visits_i, const i
 template <class Type>
 Eigen::SparseMatrix<Type> get_select_matrix(const std::vector<int>& visits_i, const int& n_visits) {
   Eigen::SparseMatrix<Type> result(visits_i.size(), n_visits);
-  for (int i = 0; i < visits_i.size(); i++) {
+  for (std::size_t i = 0, max = visits_i.size(); i != max; ++i) {
     result.insert(i, visits_i[i]) = (Type) 1.0;
   }
   return result;

